test/integration/testpaths/testpath_reduce_vm_deployments.py

```diff
@@ -0,0 +1,2076 @@
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+""" BVT tests for Virtual Machine Life Cycle
+"""
+# Import Local Modules
+from marvin.cloudstackTestCase import cloudstackTestCase
+from marvin.cloudstackAPI import (destroyVirtualMachine,
+                                  attachIso,
+                                  detachIso,
+                                  updateServiceOffering,
+                                  scaleVirtualMachine,
+                                  rebootVirtualMachine,
+                                  changeServiceForVirtualMachine)
+from marvin.lib.utils import (cleanup_resources,
+                              validateList,
+                              random_gen,
+                              is_snapshot_on_nfs,
+                              isAlmostEqual)
+from marvin.lib.base import (Account,
+                             DiskOffering,
+                             AffinityGroup,
+                             Snapshot,
+                             Volume,
+                             ServiceOffering,
+                             VirtualMachine,
+                             Host,
+                             Iso,
+                             Router,
+                             Configurations)
+from marvin.lib.common import (get_domain,
+                               list_volumes,
+                               list_hosts,
+                               list_service_offering,
+                               list_virtual_machines,
+                               list_snapshots,
+                               get_zone,
+                               get_template)
+from marvin.codes import FAILED, PASS
+from marvin.sshClient import SshClient
+from nose.plugins.attrib import attr
+
+# Import System modules
+import time
+import re
+import random
+import string
+
+_multiprocess_shared_ = True
+
+
+class TestDeployVMandVMLifeCycle(cloudstackTestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        testClient = super(TestDeployVMandVMLifeCycle, cls).getClsTestClient()
+        cls.apiclient = testClient.getApiClient()
+        cls.services = testClient.getParsedTestDataConfig()
+        cls.hypervisor = testClient.getHypervisorInfo()
+
+        # Get Zone, Domain and templates
+        domain = get_domain(cls.apiclient)
+        cls.zone = get_zone(cls.apiclient, cls.testClient.getZoneForTests())
+        cls.services['mode'] = cls.zone.networktype
+        cls.domain = get_domain(cls.apiclient)
+
+        cls.mgtSvrDetails = cls.config.__dict__["mgtSvr"][0].__dict__
```

Comment:
You are accessing private variables, do you need management server details some where in the file , I don't see any use ? Can you please let me know if i missed it somewhere ?